core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala

```diff
@@ -444,6 +444,29 @@ class WorkspaceServiceSpec extends FlatSpec with ScalatestRouteTest with Matcher
     }
   }
 
+  it should "lock a workspace with terminated submissions" in withTestDataServices { services =>
+    //check workspace is not locked
+    assert(!testData.workspaceTerminatedSubmissions.isLocked)
+
+    //lock workspace
+    services.workspaceService.lockWorkspace(new WorkspaceName(testData.workspaceTerminatedSubmissions.namespace, testData.workspaceTerminatedSubmissions.name))
```

Comment:
- await on this future otherwise the next line may potentially run before the future is executed / completed
- unpack the response and see if it's actually a 409 Conflict (or at least intercept a RawlsException)